### ISSUE TITLE
Investigate using an overlay without pulling the whole Buildroot source tree in

### ISSUE THREAD

**Issue by voltagex** [gamaral/rpi-buildroot#54]
See https://github.com/baxterworks-build/buildroot-skeleton - just an idea. Will make upgrading buildroot easier in future.

**Comment by gamaral** [comment]
Interesting idea, know if it supports packages and not just configs?

**Comment by voltagex** [comment]
Yep, use $BR2_EXTERNAL. Happy to help with this if you can show me where your customisations have been made.

**Comment by mbrgm** [comment]
@voltagex As I'd also be interested in having an up-to-date version of buildroot, I started by diffing the last merged version of buildroot with the current state of this repo:

This is the result I get from `git diff aaf6c28..HEAD`:

```patch
diff --git a/README.md b/README.md
new file mode 100644
index 0000000..25d3ee3
--- /dev/null
+++ b/README.md
@@ -0,0 +1,121 @@
+Buildroot for Raspberry Pi
+==========================
+
+This buildroot *overlay* will produce a bleeding-edge, light-weight and trimmed
+down toolchain, rootfs and kernel for the Raspberry Pi. It's intended for
+**advanced users** and specific embedded applications.
+
+Before You Begin
+----------------
+
+- If you're not familiar with Buildroot and what it can and can't do, please
+  take the time to [read the manual](http://buildroot.org/downloads/manual/manual.html).
+
+- You must be pretty comfortable with **cross-compilation** in order to use
+  rpi-buildroot.
+
+Test Drive
+----------
+
+You can test drive rpi-buildroot by following the instructions below:
+
+	wget http://dl.guillermoamaral.com/rpi/sdcard.img.xz
+	xz -d sdcard.img.xz
+	sudo dd if=sdcard.img of=/dev/sdx # replace *sdx* with your actual sdcard device node
+
+There's now also a test drive image for the Raspberry Pi 2:
+
+	wget http://dl.guillermoamaral.com/rpi/sdcard2.img.xz
+	xz -d sdcard2.img.xz
+	sudo dd if=sdcard2.img of=/dev/sdx # replace *sdx* with your actual sdcard device node
+
+The default user is **root**, no password is required.
+
+Building
+--------
+
+	git clone --depth 1 git://github.com/gamaral/rpi-buildroot.git
+	cd rpi-buildroot
+	make raspberrypi_defconfig # if your target is a Raspberry Pi 2, use 'raspberrypi2_defconfig'
+	make nconfig         # if you want to add packages or fiddle around with it
+	make                 # build (NOTICE: Don't use the **-j** switch, it's set to auto-detect)
+
+Deploying
+---------
+
+### Script
+
+I've added a script that can automatically flash your sdcard, you simply need
+to point it to the correct device node, confirm and you're done!
+
+**Notice** you will need to replace *sdx* in the following commands with the
+actual device node for your sdcard.
+
+    # run the following as root (sudo)
+    board/raspberrypi/mksdcard /dev/sdx
+
+### Manual
+
+You will need to create two partitions in your sdcard, the first (boot) needs
+to be a small *W95 FAT16 (LBA)* patition (that's partition id **e**), about 32
+MB will do.
+
+**Notice** you will need to replace *sdx* in the following commands with the
+actual device node for your sdcard.
+
+Create the partitions on the SD card. Run the following as root.
+**Notice** all data on the SD card will be lost.
+
+	fdisk /dev/sdx
+	> p             # prints partition table
+	> d             # repeat until all partitions are deleted
+	> n             # create a new partition
+	> p             # create primary
+	> 1             # make it the first partition
+	> <enter>       # use the default sector
+	> +32M          # create a boot partition with 32MB of space
+	> n             # create rootfs partition
+	> p
+	> 2
+	> <enter>
+	> <enter>       # fill the remaining disk, adjust size to fit your needs
+	> t             # change partition type
+	> 1             # select first partition
+	> e             # use type 'e' (FAT16)
+	> a             # make partition bootable
+	> 1             # select first partition
+	> p             # double check everything looks right
+	> w             # write partition table to disk.
+
+Now format the boot partition as FAT 16
+
+	# run the following as root
+	mkfs.vfat -F16 -n BOOT /dev/sdx1
+	mkdir -p /media/boot
+	mount /dev/sdx1 /media/boot
+
+You will need to copy all the files in *output/images/boot* to your *boot*
+partition.
+
+	# run the following as root
+	cp output/images/rpi-firmware/bootcode.bin /media/boot
+	cp output/images/rpi-firmware/fixup.dat /media/boot
+	cp output/images/rpi-firmware/start.elf /media/boot
+	cp output/images/zImage /media/boot/kernel.img
+	cp output/images/*.dtb /media/boot
+	umount /media/boot
+
+The second (rootfs) can be as big as you want, but with a 200 MB minimum,
+and formated as *ext4*.
+
+	# run the following as root
+	mkfs.ext4 -L rootfs /dev/sdx2
+	mkdir -p /media/rootfs
+	mount /dev/sdx2 /media/rootfs
+
+You will need to extract *output/images/rootfs.tar* onto the partition, as **root**.
+
+	# run the following as root
+	tar -xvpsf output/images/rootfs.tar -C /media/rootfs # replace with your mount directory
+	umount /media/rootfs
+
diff --git a/board/raspberrypi/genimage-raspberrypi.cfg b/board/raspberrypi/genimage-raspberrypi.cfg
index 74758f5..83eb307 100644
--- a/board/raspberrypi/genimage-raspberrypi.cfg
+++ b/board/raspberrypi/genimage-raspberrypi.cfg
@@ -5,7 +5,6 @@ image boot.vfat {
       "bcm2708-rpi-b-plus.dtb",
       "bcm2708-rpi-cm.dtb",
       "rpi-firmware/bootcode.bin",
-      "rpi-firmware/cmdline.txt",
       "rpi-firmware/config.txt",
       "rpi-firmware/fixup.dat",
       "rpi-firmware/start.elf",
diff --git a/board/raspberrypi/genimage-raspberrypi2.cfg b/board/raspberrypi/genimage-raspberrypi2.cfg
index 443c821..1441e95 100644
--- a/board/raspberrypi/genimage-raspberrypi2.cfg
+++ b/board/raspberrypi/genimage-raspberrypi2.cfg
@@ -3,7 +3,6 @@ image boot.vfat {
     files = {
       "bcm2709-rpi-2-b.dtb",
       "rpi-firmware/bootcode.bin",
-      "rpi-firmware/cmdline.txt",
       "rpi-firmware/config.txt",
       "rpi-firmware/fixup.dat",
       "rpi-firmware/start.elf",
diff --git a/board/raspberrypi/mkimage b/board/raspberrypi/mkimage
new file mode 100755
index 0000000..788a49e
--- /dev/null
+++ b/board/raspberrypi/mkimage
@@ -0,0 +1,154 @@
+#!/bin/sh
+#
+# SD card image builder for rpi-buildroot
+#
+# Guillermo A. Amaral B. <g@maral.me>
+#
+
+# sanaty check
+USERID=`id -u`
+if [ ${USERID} -ne 0 ]; then
+	echo "${0} requires root privileges in order to work."
+	exit 0
+fi
+
+section() {
+	echo "*****************************************************************************************"
+	echo "> ${*}"
+	echo "*****************************************************************************************"
+	sleep 1
+}
+
+# overrides
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+IMAGE="sdcard.img"
+OUTPUT_PREFIX=""
+
+if [ -z "${SIZE}" ]; then
+    SIZE="128M"
+fi
+
+echo "Creating ${SIZE} size image..."
+sleep 3
+
+# dependencies
+CP=`which cp`
+DD=`which dd`
+FDISK=`which fdisk`
+LOSETUP=`which losetup`
+MKDIR=`which mkdir`
+MKFS_EXT4=`which mkfs.ext4`
+MKFS_VFAT=`which mkfs.vfat`
+MOUNT=`which mount`
+RMDIR=`which rmdir`
+SYNC=`which sync`
+TAR=`which tar`
+UMOUNT=`which umount`
+
+if [ -z "${CP}" ] ||
+   [ -z "${DD}" ] ||
+   [ -z "${FDISK}" ] ||
+   [ -z "${LOSETUP}" ] ||
+   [ -z "${MKDIR}" ] ||
+   [ -z "${MKFS_EXT4}" ] ||
+   [ -z "${MKFS_VFAT}" ] ||
+   [ -z "${MOUNT}" ] ||
+   [ -z "${RMDIR}" ] ||
+   [ -z "${SYNC}" ] ||
+   [ -z "${TAR}" ] ||
+   [ -z "${UMOUNT}" ]; then
+	echo "Missing dependencies:\n"
+	echo "CP=${CP}"
+	echo "DD=${DD}"
+	echo "FDISK=${FDISK}"
+	echo "LOSETUP=${LOSETUP}"
+	echo "MKDIR=${MKDIR}"
+	echo "MKFS_EXT4=${MKFS_EXT4}"
+	echo "MKFS_VFAT=${MKFS_VFAT}"
+	echo "MOUNT=${MOUNT}"
+	echo "RMDIR=${RMDIR}"
+	echo "SYNC=${SYNC}"
+	echo "TAR=${TAR}"
+	echo "UMOUNT=${UMOUNT}"
+	exit 1
+fi
+
+# sanity check
+if [ ! -f "images/rootfs.tar" ]; then
+	if [ -f "output/images/rootfs.tar" ]; then
+		OUTPUT_PREFIX="output/"
+	else
+		echo "Didn't find boot and/or rootfs.tar! ABORT."
+		exit 1
+	fi
+fi
+
+# find loop device
+LOOP=`${LOSETUP} -f`
+
+# create image
+${DD} if=/dev/zero of=${IMAGE} bs=${SIZE} count=1
+
+# partition image
+${FDISK} ${IMAGE} <<END
+o
+n
+p
+1
+
++32M
+n
+p
+2
+
+
+t
+1
+e
+a
+1
+w
+END
+
+# loop image
+${LOSETUP} -P ${LOOP} ${IMAGE} || exit 1
+
+# format partitions
+
+section "Formatting partitions..."
+
+${MKFS_VFAT} -F 16 -n BOOT -I "${LOOP}p1" || exit 1
+${MKFS_EXT4} -F -q -L rootfs "${LOOP}p2" || exit 1
+
+${MKDIR} .mnt
+
+# fill boot
+
+section "Populating boot partition..."
+
+${MOUNT} "${LOOP}p1" .mnt || exit 2
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/bootcode.bin .mnt
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/fixup.dat .mnt
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/start.elf .mnt
+${CP} ${OUTPUT_PREFIX}images/zImage .mnt/kernel.img
+${CP} ${OUTPUT_PREFIX}images/*.dtb .mnt
+${SYNC}
+${UMOUNT} .mnt
+
+# fill rootfs
+
+section "Populating rootfs partition..."
+
+${MOUNT} "${LOOP}p2" .mnt || exit 2
+${TAR} -xpsf ${OUTPUT_PREFIX}images/rootfs.tar -C .mnt
+${SYNC}
+${UMOUNT} .mnt
+
+${RMDIR} .mnt
+
+# unmount
+${LOSETUP} -D
+
+section "Finished!"
+
+exit 0
diff --git a/board/raspberrypi/mksdcard b/board/raspberrypi/mksdcard
new file mode 100755
index 0000000..904eb9e
--- /dev/null
+++ b/board/raspberrypi/mksdcard
@@ -0,0 +1,188 @@
+#!/bin/sh
+#
+# Flash Raspberry Pi SD card [buildroot]
+#
+# Guillermo A. Amaral B. <g@maral.me>
+#
+
+SDCARD="${1}"
+
+usage() {
+	echo "Usage: ${0} [SDCARD]"
+	echo "Where SDCARD is your SD card device node, for example: /dev/sdx"
+	echo
+	echo "You will require *root* privileges in order to use this script."
+	echo
+}
+
+confirm() {
+	echo "You are about to totally decimate the following device node: ${SDCARD}"
+	echo
+	echo "If you are sure you want to continue? (Please write \"YES\" in all caps)"
+
+	read CONTUNUE
+
+	if [ "${CONTUNUE}" != "YES" ]; then
+		echo "User didn't write \"YES\"... ABORTING!"
+		exit 1
+	fi
+}
+
+section() {
+	echo "*****************************************************************************************"
+	echo "> ${*}"
+	echo "*****************************************************************************************"
+	sleep 1
+}
+
+# environment overrides
+
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+OUTPUT_PREFIX=""
+
+# check parameters
+
+if [ -z "${SDCARD}" ] || [ "${SDCARD}" = "-h" ] || [ "${SDCARD}" = "--help" ]; then
+	usage
+	exit 0
+fi
+
+# check if node is a block device
+
+if [ ! -b "${SDCARD}" ]; then
+	echo "${SDCARD} is not a block device!"
+	exit 1
+fi
+
+# root privilege check
+
+USERID=`id -u`
+if [ ${USERID} -ne 0 ]; then
+	echo "${0} requires root privileges in order to work."
+	exit 0
+fi
+
+# dependencies
+
+CP=`which cp`
+FDISK=`which fdisk`
+MKDIR=`which mkdir`
+MKFS_EXT4=`which mkfs.ext4`
+MKFS_VFAT=`which mkfs.vfat`
+MOUNT=`which mount`
+RMDIR=`which rmdir`
+SYNC=`which sync`
+TAR=`which tar`
+UMOUNT=`which umount`
+
+if [ -z "${CP}" ] ||
+   [ -z "${FDISK}" ] ||
+   [ -z "${MKDIR}" ] ||
+   [ -z "${MKFS_EXT4}" ] ||
+   [ -z "${MKFS_VFAT}" ] ||
+   [ -z "${MOUNT}" ] ||
+   [ -z "${RMDIR}" ] ||
+   [ -z "${TAR}" ] ||
+   [ -z "${UMOUNT}" ]; then
+	echo "Missing dependencies:\n"
+	echo "CP=${CP}"
+	echo "FDISK=${FDISK}"
+	echo "MKDIR=${MKDIR}"
+	echo "MKFS_EXT4=${MKFS_EXT4}"
+	echo "MKFS_VFAT=${MKFS_VFAT}"
+	echo "MOUNT=${MOUNT}"
+	echo "RMDIR=${RMDIR}"
+	echo "TAR=${TAR}"
+	echo "UMOUNT=${UMOUNT}"
+	exit 1
+fi
+
+# sanity check
+
+if [ ! -d "images/rpi-firmware" ] || [ ! -f "images/rootfs.tar" ]; then
+	if [ -d "output/images/rpi-firmware" ] && [ -f "output/images/rootfs.tar" ]; then
+		OUTPUT_PREFIX="output/"
+	else
+		echo "Didn't find rpi-firmware and/or rootfs.tar! ABORT."
+		exit 1
+	fi
+fi
+
+# warn user
+
+confirm
+
+# partition image
+
+section "Partitioning SD card..."
+
+${FDISK} ${SDCARD} <<END
+o
+n
+p
+1
+
++32M
+n
+p
+2
+
+
+t
+1
+e
+a
+1
+w
+END
+
+sleep 1
+
+# figure out partition pattern
+
+case "${SDCARD}" in
+	*mmcblk*) SDCARDP="${SDCARD}p" ;;
+	*)        SDCARDP="${SDCARD}"  ;;
+esac
+
+# format partitions
+
+section "Formatting partitions..."
+
+${MKFS_VFAT} -F 16 -n BOOT -I "${SDCARDP}1" || exit 1
+${MKFS_EXT4} -F -q -L rootfs "${SDCARDP}2" || exit 1
+
+
+# prepare to fill partitions
+
+${MKDIR} .mnt
+
+# fill boot
+
+section "Populating boot partition..."
+${MOUNT} "${SDCARDP}1" .mnt || exit 2
+
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/bootcode.bin .mnt
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/fixup.dat .mnt
+${CP} ${OUTPUT_PREFIX}images/rpi-firmware/start.elf .mnt
+${CP} ${OUTPUT_PREFIX}images/zImage .mnt/kernel.img
+${CP} ${OUTPUT_PREFIX}images/*.dtb .mnt
+${SYNC}
+${UMOUNT} .mnt
+
+# fill rootfs
+
+section "Populating rootfs partition..."
+${MOUNT} "${SDCARDP}2" .mnt || exit 2
+
+${TAR} -xpsf ${OUTPUT_PREFIX}images/rootfs.tar -C .mnt
+${SYNC}
+${UMOUNT} .mnt
+
+# clean up
+
+${RMDIR} .mnt
+
+section "Finished!"
+
+exit 0
diff --git a/board/raspberrypi/overlay/etc/fstab b/board/raspberrypi/overlay/etc/fstab
new file mode 100644
index 0000000..0dd2a16
--- /dev/null
+++ b/board/raspberrypi/overlay/etc/fstab
@@ -0,0 +1,10 @@
+# /etc/fstab: static file system information.
+#
+# <file system> <mount pt> <type> <options>               <dump> <pass>
+/dev/root       /          ext4   rw,noauto               0      1
+proc            /proc      proc   defaults                0      0
+devpts          /dev/pts   devpts defaults,gid=5,mode=620 0      0
+tmpfs           /dev/shm   tmpfs  mode=0777               0      0
+tmpfs           /tmp       tmpfs  mode=1777               0      0
+tmpfs           /run       tmpfs  mode=0755,nosuid,nodev  0      0
+sysfs           /sys       sysfs  defaults                0      0
diff --git a/board/raspberrypi/overlay/etc/issue b/board/raspberrypi/overlay/etc/issue
new file mode 100644
index 0000000..9e87483
--- /dev/null
+++ b/board/raspberrypi/overlay/etc/issue
@@ -0,0 +1,7 @@
+[H[2J┌──────────────────────────────────────────────┐
+│[0;1;35;95m░█[0;1;31;91m▀▄[0;1;33;93m░█[0;1;32;92m▀█[0;1;36;96m░█[0;1;34;94m▀▀[0;1;35;95m░█[0;1;31;91m▀█[0;1;33;93m░█[0;1;32;92m▀▄[0;1;36;96m░█[0;1;34;94m▀▀[0;1;35;95m░█[0;1;31;91m▀▄[0;1;33;93m░█[0;1;32;92m▀▄[0;1;36;96m░█[0;1;34;94m░█[0;1;35;95m░░[0;1;31;91m░█[0;1;33;93m▀█[0;1;32;92m░▀[0;1;36;96m█▀[0m│
+│[0;1;31;91m░█[0;1;33;93m▀▄[0;1;32;92m░█[0;1;36;96m▀█[0;1;34;94m░▀[0;1;35;95m▀█[0;1;31;91m░█[0;1;33;93m▀▀[0;1;32;92m░█[0;1;36;96m▀▄[0;1;34;94m░█[0;1;35;95m▀▀[0;1;31;91m░█[0;1;33;93m▀▄[0;1;32;92m░█[0;1;36;96m▀▄[0;1;34;94m░░[0;1;35;95m█░[0;1;31;91m░░[0;1;33;93m░█[0;1;32;92m▀▀[0;1;36;96m░░[0;1;34;94m█░[0m│
+│[0;1;33;93m░▀[0;1;32;92m░▀[0;1;36;96m░▀[0;1;34;94m░▀[0;1;35;95m░▀[0;1;31;91m▀▀[0;1;33;93m░▀[0;1;32;92m░░[0;1;36;96m░▀[0;1;34;94m▀░[0;1;35;95m░▀[0;1;31;91m▀▀[0;1;33;93m░▀[0;1;32;92m░▀[0;1;36;96m░▀[0;1;34;94m░▀[0;1;35;95m░░[0;1;31;91m▀░[0;1;33;93m░░[0;1;32;92m░▀[0;1;36;96m░░[0;1;34;94m░▀[0;1;35;95m▀▀[0m│
+└──────────────────────────────────────────────┘
+
+Your slice of Raspberry Pi is now ready!
diff --git a/configs/raspberrypi2_defconfig b/configs/raspberrypi2_defconfig
index ce41ee3..0e481ab 100644
--- a/configs/raspberrypi2_defconfig
+++ b/configs/raspberrypi2_defconfig
@@ -5,19 +5,24 @@ BR2_ARM_FPU_NEON_VFPV4=y
 
 BR2_TOOLCHAIN_BUILDROOT_CXX=y
 
-BR2_TARGET_GENERIC_GETTY_PORT="tty1"
+BR2_ROOTFS_OVERLAY="board/raspberrypi/overlay"
+BR2_TARGET_GENERIC_GETTY_PORT="ttyAMA0"
+BR2_TARGET_GENERIC_HOSTNAME="rpi"
+BR2_TARGET_GENERIC_ISSUE=""
 
+BR2_KERNEL_HEADERS_VERSION=y
+BR2_DEFAULT_KERNEL_VERSION="4.1.14"
 # Linux headers same as kernel, a 4.1 series
 BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_1=y
 
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_CUSTOM_GIT=y
-BR2_LINUX_KERNEL_CUSTOM_REPO_URL="https://github.com/raspberrypi/linux.git"
-BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="d33d0293e245badc4ca6ede3984d8bb8ea63cb1a"
+BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://github.com/raspberrypi/linux.git"
+BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="rpi-4.1.y"
 BR2_LINUX_KERNEL_DEFCONFIG="bcm2709"
+BR2_LINUX_KERNEL_USE_DEFCONFIG=y
 BR2_LINUX_KERNEL_ZIMAGE=y
 
-# Build the DTB from the kernel sources
 BR2_LINUX_KERNEL_DTS_SUPPORT=y
 BR2_LINUX_KERNEL_INTREE_DTS_NAME="bcm2709-rpi-2-b"
 
diff --git a/configs/raspberrypi_defconfig b/configs/raspberrypi_defconfig
index 3b3cc50..809bbce 100644
--- a/configs/raspberrypi_defconfig
+++ b/configs/raspberrypi_defconfig
@@ -1,19 +1,26 @@
 BR2_arm=y
 BR2_arm1176jzf_s=y
 BR2_ARM_EABIHF=y
+BR2_ARM_FPU_VFPV2=y
 
-BR2_TARGET_GENERIC_GETTY_PORT="tty1"
+BR2_TOOLCHAIN_BUILDROOT_CXX=y
+
+BR2_ROOTFS_OVERLAY="board/raspberrypi/overlay"
+BR2_TARGET_GENERIC_GETTY_PORT="ttyAMA0"
+BR2_TARGET_GENERIC_HOSTNAME="rpi"
+BR2_TARGET_GENERIC_ISSUE=""
 
+BR2_KERNEL_HEADERS_VERSION=y
+BR2_DEFAULT_KERNEL_VERSION="4.1.14"
 # Linux headers same as kernel, a 4.1 series
 BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_1=y
 
-BR2_TOOLCHAIN_BUILDROOT_CXX=y
-
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_CUSTOM_GIT=y
-BR2_LINUX_KERNEL_CUSTOM_REPO_URL="https://github.com/raspberrypi/linux.git"
-BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="d33d0293e245badc4ca6ede3984d8bb8ea63cb1a"
+BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://github.com/raspberrypi/linux.git"
+BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="rpi-4.1.y"
 BR2_LINUX_KERNEL_DEFCONFIG="bcmrpi"
+BR2_LINUX_KERNEL_USE_DEFCONFIG=y
 BR2_LINUX_KERNEL_ZIMAGE=y
 
 # Build the DTBs for A/B, A+/B+ and compute module from the kernel sources
diff --git a/package/rpi-firmware/rpi-firmware.mk b/package/rpi-firmware/rpi-firmware.mk
index 4b92a50..4846297 100644
--- a/package/rpi-firmware/rpi-firmware.mk
+++ b/package/rpi-firmware/rpi-firmware.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 22d8c910f4c53118f9cf85c038d7d8e307efc110
+RPI_FIRMWARE_VERSION = 1.20160309
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3c
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
@@ -41,7 +41,6 @@ define RPI_FIRMWARE_INSTALL_IMAGES_CMDS
 	$(INSTALL) -D -m 0644 $(@D)/boot/start$(BR2_PACKAGE_RPI_FIRMWARE_BOOT).elf $(BINARIES_DIR)/rpi-firmware/start.elf
 	$(INSTALL) -D -m 0644 $(@D)/boot/fixup$(BR2_PACKAGE_RPI_FIRMWARE_BOOT).dat $(BINARIES_DIR)/rpi-firmware/fixup.dat
 	$(INSTALL) -D -m 0644 package/rpi-firmware/config.txt $(BINARIES_DIR)/rpi-firmware/config.txt
-	$(INSTALL) -D -m 0644 package/rpi-firmware/cmdline.txt $(BINARIES_DIR)/rpi-firmware/cmdline.txt
 	$(RPI_FIRMWARE_INSTALL_DTB)
 	$(RPI_FIRMWARE_INSTALL_DTB_OVERLAYS)
 endef
```

**Comment by MrAda** [comment]
I think gamaral is on vacation.  I hope when he gets back we can get some help with this.  I have yet to get a build root creation to work on my RPi3

**Comment by gamaral** [comment]
RPis are pretty well supported in vanilla Buildroot now. Images get created and everything. ;-) I don't really think you guys need to go through the trouble. ;-)

**Comment by MrAda** [comment]
That isn't true.  It only supports HDMI output, I don't have a device to output on unless maybe my TV.  I want to get it to boot fast and then work on the driver for the LCD panel I have, for which I have no technical information.

**Comment by gamaral** [comment]
If you need TTY to be on the UART, you can change that in the standard config.

I.e. BR2_TARGET_GENERIC_GETTY_PORT="ttyAMA0"

**Comment by voltagex** [comment]
@mbrgm can you trim that post and use attachments or gists for that amount of text? It's making the discussion hard to follow

**Comment by MrAda** [comment]
@gamaral I don't understand.  What does TTY or UART have to do with the LCD screen?  That sounds like a keyboard kind of interface.  The raspberry pi official 7" screen with touch is connect to the Pi by an DSI ribbon cable.

**Comment by gamaral** [comment]
Maybe I misunderstood "That isn't true. It only supports HDMI output". Did you mean it doesn't support RCA then?

**Comment by MrAda** [comment]
No i didn't mean RCA, didn't realize there was an RCA out.  What I have is the official Raspberry pi LCD display and it has an DSI ribbon connector.

**Comment by voltagex** [comment]
Hasn't this gone a bit off topic now?

**Comment by gamaral** [comment]
I have to agree. In any case, like I said, most changes in this report are no longer required to get a system up and working. I would say stick with Buildroot proper instead. ;-)

**Comment by gamaral** [comment]
Sorry, not report, repo. Autocorrect. :-(

**Comment by voltagex** [comment]
Closing this as I'm no longer using this code.
